### PR TITLE
Feature/test user exporter

### PR DESCRIPTION
**Pull request by joerghoh**
Added test coverage for the user exporter servlets.

NOTE: I had to change the JCR Queries which are used to find users and groups. It should not make any difference in the result, but the runtime might differ (using indexes or not).

Nevertheless, these queries should be replaced by proper calls to the JCR Usermanager which internally handles this much better than these hand-crafted queries ... 
But that's something for the next PR.

(When merging, consider to squash this PR!)

### CODE REVIEW

File: bundle/pom.xml [hunk 1]
```diff
@@ -443,6 +443,8 @@
             <groupId>com.github.ben-manes.caffeine</groupId>
             <artifactId>caffeine</artifactId>
         </dependency>
+
```
Comment (davidjgonzalez) [inline bundle/pom.xml]: Remove empty lines

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+public class UserExportServletTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UserExportServletTest.class);
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java]: Dont need LOG in Test class

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    
+    @Test
+    public void testWithGroupDirectFilter() throws Exception {
+
+        // Build parameters
+        JsonObject params = buildParameterObject("direct", "users");
+
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("params", params);
+        LOG.info("params = {}",params.toString());
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java]: No LOG

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+        
+        context.request().setParameterMap(parameters);
+        servlet.doGet(context.request(), context.response());
+        
+        assertEquals(context.response().getStatus(), 200);
+        String output = context.response().getOutputAsString();
+        LOG.info(output);
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java]: No LOG

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+public class UserExportServletTest {
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java]: This test class has a bunch of unnecessary whitespace/empty lines - this should be collapse/formatted appropriately.

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+        servlet = new UsersExportServlet();
+    }
+
+    @Test
+    public void testWithNoParameterProvidedInRequest() throws Exception {
+        servlet.doGet(context.request(), context.response());
+        assertEquals(context.response().getStatus(), 200);
+        String output = context.response().getOutputAsString();
+        LOG.info("output = {}",output);
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UserExportServletTest.java]: No need to LOG during test

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersInitServletTest.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class UsersInitServletTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UsersInitServletTest.class);
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersInitServletTest.java]: No LOG

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersInitServletTest.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+import org.junit.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class UsersInitServletTest {
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersInitServletTest.java]: This test class also has a bunch of unnecessary whitespace/empty lines - this should be collapse/formatted appropriately.

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersInitServletTest.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+        
+        
+    }
+    
+    
+    @Test
+    public void testGroupExistence() throws Exception {
+        UsersInitServlet servlet = new UsersInitServlet();
+        servlet.doGet(context.request(), context.response());
+        LOG.info(context.response().getOutputAsString());
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersInitServletTest.java]: No LOG

File: bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersSaveServletTest.java [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+import java.util.Map;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.testing.mock.sling.junit.SlingContext;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class UsersSaveServletTest {
```
Comment (davidjgonzalez) [inline bundle/src/test/java/com/adobe/acs/commons/exporters/impl/users/UsersSaveServletTest.java]: This test class has a bunch of unnecessary whitespace/empty lines - this should be collapse/formatted appropriately.

**Comment by joerghoh** [comment]
@davidjgonzalez here you are

**Comment by badvision** [comment]
50% coverage!  WOO HOO @joerghoh! 😄

**Comment by joerghoh** [comment]
yay!
